sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/InMemoryTableScanExec.scala

```diff
@@ -102,6 +102,7 @@ case class InMemoryTableScanExec(
     case IsNull(a: Attribute) => statsFor(a).nullCount > 0
     case IsNotNull(a: Attribute) => statsFor(a).count - statsFor(a).nullCount > 0
 
+    case In(_: AttributeReference, list: Seq[Expression]) if list.isEmpty => Literal.FalseLiteral
```

Comment:
This should be moved to optimizer rules

Reply:
BTW, this is wrong. If the left value is null, it should return null instead of false.

Reply:
Ah, really. OK, yeah we need to change this. It can be reverted too.

Reply:
since this is only for filters, does it make any difference null or false?

Reply:
Under this special context of filtering partitions, this `In` with empty list will result in a false literal in the end, no matter the attribute is null or not. We don't possibly have some expressions like `IsNull(In(a, Nil))` as the filter predicate for now.